test/unit/OrchestratorUnitTests.cpp

```diff
@@ -280,64 +285,71 @@ TEST_CASE_METHOD(
             {
                 .IsSubscribe = true,
                 .ChannelId = channelId,
-                .StreamKey = std::vector<std::byte>(),
+                .StreamKey = streamKey,
             });
     }
 
     // Connect the ingest and have it report the stream
     auto ingest = generateAndConnectMockConnection("ingest");
-    ingest->MockFireOnStreamPublish(
+    std::vector<ConnectionRelayPayload> recvRelayPayloads;
+    ingest->SetOnStreamRelay(
+        [&recvRelayPayloads](ConnectionRelayPayload payload)
```

Comment:
Won't this be invoked async, meaning `recvRelayPayloads` will not be populated in time for the use of it on like 315?

Reply:
Actually no, at least not yet - once `MockConnection::MockFireOnStreamPublish` is fired below, the callback to `Orchestrator::connectionStreamPublish` is called, and `Orchestrator` sends out relay messages, all on the same thread, and those end up hitting this lambda. 

Whether it _ought_ to be async is another conversation :D